controller/controller.go

```diff
@@ -52,6 +52,11 @@ type Controller struct {
 }
 
 func NewController(cfg *Config) (*Controller, error) {
+	xt.GlobalRegistry().RegisterFactory(xt_smartrouting.NewFactory())
+	xt.GlobalRegistry().RegisterFactory(xt_ha.NewFactory())
+	xt.GlobalRegistry().RegisterFactory(xt_random.NewFactory())
+	xt.GlobalRegistry().RegisterFactory(xt_weighted.NewFactory())
+
```

Comment:
Maybe we could stick these in a `registerXts` or `registerTerminators`, similar to `registerComponents` in the same file?